### PR TITLE
Add tests of service

### PR DESCRIPTION
**Pull request by silemie**
Fix 34, 40-42, 50-52, 60-62, 70-72, 133-134, 173, 179-182, 196, 212-219

service/service.py    85    2    98%   135-136

### CODE REVIEW

**Comment by lhmartin** [comment]
I fixed it locally and the test run fine on my end, good job!

**Comment by silemie** [comment]
@lhmartin The bug was fixed in new version

**Comment by lhmartin** [comment]
for the "Test Internal Error Handle " the response coming back is a 415 not a 500.

**Comment by silemie** [comment]
Solve it! Final result:
image

File: service/service.py [hunk 1]
```diff
@@ -168,19 +174,9 @@ def action_recommend_product(productId):
...
     app.logger.info('Recommend product')
     suppliers = Supplier.action_make_recommendation(productId)
-    if suppliers is not None:
+    if len(suppliers) > 0:
         return make_response(suppliers.to_json(), status.HTTP_200_OK)
-    return make_response("NOT FOUND", status.HTTP_404_NOT_FOUND)
-
-
-@app.route('/suppliers', methods=['GET'])
-def query_a_supplier():
```
Comment (greenson) [inline service/service.py]: Query function missing?

File: service/service.py [hunk 2]
```diff
@@ -212,11 +208,7 @@ def init_db():
 
 def check_content_type(content_type):
     """ Checks whether the request content type is correct """
-    try:
```
Comment (greenson) [inline service/service.py]: Null Content-Type validation missing?

File: service/service.py (hunk 1 above)
Comment (silemie) [reply service/service.py]: I put query operation into list operation. 
Like the professor's code..

Comment (greenson) [reply service/service.py]: OK cool!

**Pull request merged**